src/openms/include/OpenMS/KERNEL/SpectrumRangeManager.h

```diff
@@ -0,0 +1,186 @@
+      
+      @param other The other range manager to extend from
+      @param ms_level The MS level for which to extend the ranges (0 for global ranges)
+    */
+    void extend(const BaseType& other, UInt ms_level = 0)
+    {
+      if (ms_level == 0)
+      {
+        BaseType::extendUnsafe(other);
+      }
+      else
+      {
+        if (ms_level_ranges_.find(ms_level) == ms_level_ranges_.end())
+        {
+          ms_level_ranges_[ms_level] = other;
+        }
+        else
+        {
+          ms_level_ranges_[ms_level].extendUnsafe(other);
+        }
+      }
+    }
+    
+    /**
+      @brief Gets the ranges for a specific MS level
+      
+      @param ms_level The MS level for which to retrieve the ranges (0 for global ranges)
+      @return The ranges for the specified MS level (returns the global ranges if no specific ranges exist for the requested MS level)
+    */
+    const BaseType& operator()(UInt ms_level = 0) const
+    {
+      if (ms_level == 0 || ms_level_ranges_.find(ms_level) == ms_level_ranges_.end())
+      {
+        return *this;
+      }
+      return ms_level_ranges_.at(ms_level);
+    }
+
+    /**
+      @brief Gets all MS levels for which specific ranges exist
+      
+      @return The set of MS levels
+    */
+    std::set<UInt> getMSLevels() const
+    {
+      std::set<UInt> ms_levels;
+      for (const auto& [level, _] : ms_level_ranges_)
+      {
+        ms_levels.insert(level);
+      }
+      return ms_levels;
+    }
+
+    /**
+      @brief Extends the RT range with an MS level parameter
+      
+      @param rt The RT value to extend with
+      @param ms_level The MS level for which to extend the RT range (0 for global range)
+    */
+    void extendRT(double rt, UInt ms_level)
+    {
+      if (ms_level == 0)
+      {
+        BaseType::extendRT(rt);
+      }
+      else
+      {
+        if (ms_level_ranges_.find(ms_level) == ms_level_ranges_.end())
+        {
+          ms_level_ranges_[ms_level] = BaseType();
+        }
+        ms_level_ranges_[ms_level].extendRT(rt);
+      }
+    }
+
+    /**
+      @brief Extends the RT range with an MS level parameter
+      
+      @param rt The RT value to extend with
+      @param ms_level The MS level for which to extend the RT range (0 for global range)
+    */
+    void extendMZ(double rt, UInt ms_level)
+    {
+      if (ms_level == 0)
+      {
+        BaseType::extendMZ(rt);
+      }
+      else
+      {
+        if (ms_level_ranges_.find(ms_level) == ms_level_ranges_.end())
+        {
+          ms_level_ranges_[ms_level] = BaseType();
+        }
+        ms_level_ranges_[ms_level].extendMZ(rt);
+      }
+    }
+
+
+    /**
+      @brief Extends the ranges with the ranges of a spectrum using an MS level parameter
+      
+      @param spectrum The spectrum whose ranges to extend from
+      @param ms_level The MS level for which to extend the ranges (0 for global ranges)
+    */
+    void extendUnsafe(const MSSpectrum& spectrum, UInt ms_level)
```

Comment:
@cbielow what is the correct way to extend this OpenMS::RangeManager<OpenMS::RangeMZ, OpenMS::RangeIntensity, OpenMS::RangeMobility, OpenMS::RangeRT> with a OpenMS::RangeManager<OpenMS::RangeMZ, OpenMS::RangeIntensity, OpenMS::RangeMobility>

Reply:
ah found it